pkg/controller/completed_workflows.go

```diff
@@ -73,15 +75,26 @@ func CalculateHoursToDelete(retentionPeriodHours, currentHourOfDay int) []string
 	return hoursToDelete
 }
 
+//Calculates a list of all the hours that should be kept given the current time and the retentionperiod in hours
+func CalculateHoursToKeep(retentionPeriodHours int, currentTime time.Time) []string {
+	hoursToKeep := make([]string, 0, retentionPeriodHours+1)
+	for i := 0; i <= retentionPeriodHours; i++ {
+		hoursToKeep = append(hoursToKeep, strings.ReplaceAll(
+			strings.Split(currentTime.Round(time.Hour).String(), ":")[0], " ", "."))
```

Comment:
Can you elaborate on why are we splitting by ":" here and by " +" in line 46?

Reply:
Not: maybe you can move that logic (the split and replace) to a simple function and use that in both places to make sure the behavior is always consistent